### PR TITLE
Fix DLK code generation test cases

### PR DESCRIPTION
**Pull request by hadusam**
Step 1 of #477 

I changed test cases to inherit base-file definition, use inputs of table testing and separated test files of each architecture for running each test separately.

| Task | Model | x86_64 | ARM | ARM_FPGA |
| --- | --- | --- | --- | --- |
| classification | lmnet_quantize_cifar10_space_to_depth | o | o | o |
| classification | resnet_quantize_cifar10 | o | o | o |
| object_detection | fyolo_quantize_4_v4 | o | o | o |
| segmentation | lm_segnet_v1_quantize_camvid | o | o | o |

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: dlk/tests/test_code_generation_x86_64.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+def get_configurations():
+    cpu_name = "x86_64"
+    test_cases = [
+        {'use_avx': True, 'hard_quantize': True, 'threshold_skipping': True},
+        {'use_avx': True, 'hard_quantize': True, 'threshold_skipping': False},
+        {'use_avx': True, 'hard_quantize': False, 'threshold_skipping': False},
+        {'use_avx': False, 'hard_quantize': True, 'threshold_skipping': True},
+        {'use_avx': False, 'hard_quantize': True, 'threshold_skipping': False},
+        {'use_avx': False, 'hard_quantize': False, 'threshold_skipping': False},
```
Comment (hadusam) [inline dlk/tests/test_code_generation_x86_64.py]: There are test cases for x86_64.

| use_avx | hart_quantize | threshold_skipping |
| --- | --- | --- |
| o | o | o |
| o | o | x |
| o | x | o |
| x | o | o |
| x | o | x |
| x | x | o |

There was no case of hard_quantize:`False` and threshold_skipping:`False`
I am not sure about these test cases, so I am following the previous test cases.
@iizukak 
If we need tests with the case hard_quantize:`False` and threshold_skipping:`False`, please let me know, I will add them.

Comment (iizukak) [reply dlk/tests/test_code_generation_x86_64.py]: @hadusam Thanks. We don't need Both `False` case.

Comment (hadusam) [reply dlk/tests/test_code_generation_x86_64.py]: @iizukak Thank you! Then, plz review this PR 😄

**Comment by hadusam** [comment]
@tsawada 
Plz review python readability 🙇 

Unfortunately, Github PR's `Files Changed` cannot show the diff of renamed files.
But I only have small changes to that file.
I recommend you to see the blame of `test_code_generation_base.py` to review it.
https://github.com/blue-oil/blueoil/blame/b285b98b85a350adc8a4f5a4649403382bae1124/dlk/tests/test_code_generation_base.py

Thanks.

**Comment by tsawada** [comment]
it's hard to review a diff this big. can you consider splitting?

**Comment by hadusam** [comment]
@tsawada OK, I will try it 👍

**Comment by hadusam** [comment]
@tsawada 
As you mentioned, a diff was too large. So, I focused to change to rewrite test case configurations to be generated by the function.
There are many redundant and hard-coded configurations by about 300 over lines in current master codes, and these were too complicated, so I removed them and make `4 base configuration` x `4 CPU architecture` x `test cases of each architechture`.
You can see the current test case situation on #477.

Is it reviewable? Please recheck this PR! 🙇 
Thanks! 👍

**Comment by hadusam** [comment]
@tsawada Thanks!
OK, I will separate them and create another PRs 👍

**Comment by hadusam** [comment]
This PR was duplicated by some PRs (#510, #511, #518, #520).

**Pull request closed without merging**